ext_mongodb.php

```diff
@@ -613,11 +613,90 @@ public function __construct(?array $bulkWriteOptions = array());
 	<<__Native>>
 	public function insert(mixed $document) : mixed;
 
+	private function _queryOptBool( array &$transformedOptions, array $options, string $key )
+	{
+		if ( array_key_exists( $key, $options ) )
+		{
+			$transformedOptions[$key] = (bool) $options[$key];
+		}
+	}
+
+	private function _queryOptDocument( array &$transformedOptions, array $options, string $key )
+	{
+		if ( array_key_exists( $key, $options ) )
+		{
+			if ( !is_array( $options[$key] ) && !is_object( $options[$key] ) )
+			{
+				throw new \MongoDB\Driver\Exception\InvalidArgumentException(
+					sprintf(
+						'Expected "%s" option to be array or object, %s given',
+						$key, gettype( $options[$key] )
+					)
+				);
+			}
+
+			$transformedOptions[$key] = $options[$key];
+		}
+	}
+
+	private function _transformUpdateOptions( array $options = [] ) : array
+	{
+		$transformedOptions = [];
+
+		$this->_queryOptBool( $transformedOptions, $options, 'multi' );
+		$this->_queryOptBool( $transformedOptions, $options, 'upsert' );
+
+		return $transformedOptions;
+	}
+
+	private function _updateHasOperators( array $update )
+	{
+		foreach ( $update as $key => $value )
+		{
+			if ( $key[0] == '$' )
```

Comment:
Now that I look at the native implementation for `_update()`, was there any reason to do this in PHP instead of copying the C source that `php_phongo_bulkwrite_update_has_operators()` and libmongoc both use? That would abstract this logic inside of the `_update()` call entirely from the PHP perspective.